kaggle_environments/envs/hungry_geese/hungry_geese.js

```diff
@@ -151,36 +151,53 @@ function renderer({
     ],
   };
 
-  const getRowCol = (cell) => [Math.floor(cell / columns), cell % columns];
+  // Javascript does not correctly handle mod of negatives
+  const mod = (number, divisor) => ((number % divisor) + divisor) % divisor;
+
+  const getRowCol = (cell) => [Math.floor(cell / columns), mod(cell, columns)];
+
+  const diff = (row, column, nextRow, nextColumn, rows, columns) => {
+    const north = mod(row - 1, rows) === nextRow
+    const south = mod(row + 1, rows) === nextRow
+    const east = mod(column + 1, columns) === nextColumn
+    const west = mod(column - 1, columns) === nextColumn
+    return [north, south, east, west];
+  }
 
-  // Observation.
   const { geese, food } = environment.steps[step][0].observation;
 
   // Organize the geese positions for rendering.
   const geesePositions = {};
-  geese.forEach((goose, index) => {
+  geese.forEach((goose, gooseIndex) => {
     const active = goose.length > 0;
     let s = step;
     if (!active) {
       // Find the last step where the goose existed.
       while (s >= 0 && !goose.length) {
```

Comment:
nit: Perhaps negligible, but I think this could be binary search instead?

Reply:
Fair point, but negligible utility in this case as it runs in browser and there are very few elements to traverse.